src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,250 @@
+                } else if (currentCar.getId().equals(resultSet.getLong("car_id"))) {
+                    addDriverToCar(buildDriverFromResultSet(resultSet), currentCar);
+                } else {
+                    cars.add(currentCar);
+                    currentCar = buildCarFromResultSet(resultSet);
+                }
+            }
+            cars.add(currentCar);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(updateRequest)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars DB.", e);
+        }
+        deletePreviousRelations(car.getId());
+        setCarToDriversRelation(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deletePreviousRelations(id);
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Long> carsId = new ArrayList<>();
+        String getAllByDriverRequest =
+                "SELECT cars_id "
+                        + "FROM cars_drivers cd "
+                        + "JOIN cars c ON cd.cars_id = c.id "
+                        + "WHERE drivers_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(getAllByDriverRequest)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carsId.add(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        List<Car> cars = new ArrayList<>();
+        extractCarsByCarsId(carsId, cars);
```

Comment:
Isn't it possible to get all cars in the previous try/catch block instead of car ids? Also do not forget to add all drivers to found cars